contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -273,12 +273,97 @@ def get_halfchannel(self,
         channel = self.get_channel(short_channel_id)
         return channel.half_channels[direction]
 
+    def get_halfchannels(self,
+                         source: Union[GossmapNodeId, str, None] = None,
+                         destination: Union[GossmapNodeId, str, None] = None,
+                         depth: int = 0,
+                         excludes: Union[Set[Any], List[Any]] = set()):
+        """ Returns a set[GossmapHalfchannel]` from `source` or towards
+            `destination` node ID. Using the optional `depth` greater than `0`
+            will result in a second, third, ...  order list of connected
+            channels towards or from that node.
+            Note: only one of `source` or `destination` can be given. """
+        assert (source is None) ^ (destination is None), "Only one of source or destination must be given"
+        assert depth >= 0, "Depth cannot be smaller than 0"
+        node = self.get_node(source if source else destination)
+        assert node is not None, "source or destination unknown"
+        if isinstance(excludes, List):
+            excludes = set(excludes)
+
+        # first get set of reachable nodes ...
+        reachable = self.get_nodes(source, destination, depth, excludes)
+        # and iterate and check any each source/dest channel from here
+        result = set()
+        for node in reachable:
+            for channel in node.channels:
+                if channel in excludes:
+                    continue
+                other = channel.node1 if node != channel.node1 else channel.node2
+                if other in reachable or other in excludes:
+                    continue
+                direction = 0
+                if source is not None and node > other:
+                    direction = 1
+                if destination is not None and node < other:
+                    direction = 1
+                hc = channel.half_channels[direction]
+                # skip excluded or non existent halfchannels
+                if hc is None or hc in excludes:
+                    continue
+                result.add(hc)
+        return result
+
     def get_node(self, node_id: Union[GossmapNodeId, str]):
         """ Resolves a node by its public key node_id """
         if isinstance(node_id, str):
             node_id = GossmapNodeId.from_str(node_id)
         return self.nodes.get(node_id)
 
+    def get_nodes(self,
```

Comment:
This is also a pretty big misnomer, it's more `neighbors()` than just a plain `get_nodes`.

Reply:
Yep, `get_neighbors(...)` is much better. thanks